### PR TITLE
refactor StellarGradebook code

### PR DESCRIPTION
**Pull request by pwilkins**
Refactor the StellarGradebook code into Client, Base, Membership, and Gradebook classes.  The intent of this PR is simply to separate the code into separate classes, nothing more.  If this PR is reviewed as something more comprehensive, I'd prefer to resubmit the PR so that it specifically identifies the broader goal.
Addresses #7, #8, #9, and #10.
This PR is ready for review.  @pdpinch @carsongee

### CODE REVIEW

**Comment by carsongee** [comment]
Looks like you have some pep-8 stuff to deal with

File: pylmod/gradebook.py [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+
+from base import Base
+
+
+log = logging.getLogger(__name__)
+
+
+class GradeBook(Base):
+    """
+
```
Comment (carsongee) [inline pylmod/gradebook.py]: Please add an actual docstring.  Preferably the google style class doc string, but a simple description is fine too

File: pylmod/base.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import logging
+import requests
+
+
+log = logging.getLogger(__name__)  # pylint: disable=C0103
+
+
+class Base(object):
+    """
+    Python class representing interface to Stellar gradebook.
```
Comment (carsongee) [inline pylmod/base.py]: All of this is wrong with this change, this should be the doc string for client maybe if you want to reuse, but this docstring should be something about how this is a base class that shouldn't be instantiated directly and a description of the base/common methods/attributes it provides for people wanting to extend it.

File: pylmod/membership.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
+"""
+
```
Comment (carsongee) [inline pylmod/membership.py]: Again, please actually describe the module here, something as simple as "This is the module that contains the Membship class for use with the LMOD Membership API (url to docs)"

File: pylmod/membership.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+from base import Base
+
+
+log = logging.getLogger(__name__)
+
+
+class Membership(Base):
+    """
+
```
Comment (carsongee) [inline pylmod/membership.py]: And again

**Comment by carsongee** [comment]
https://github.com/mitodl/PyLmod/blob/feature/refactor-stellargradebook/pylmod/stellargradebook.py should no longer be here, and we should see it being removed from the diff.  It has the added benefit that we can see better which pieces got moved where.

**Comment by pwilkins** [comment]
In my enthusiasm to get this committed I left numerous pylint and pep8 errors.  I'm working on those now.  I really only wanted to verify that code is split up properly for this PR.

**Comment by carsongee** [comment]
No problem, it is just hard to tell without stellargradebook.py being deleted if you could commit that real quick I can ignore the style problems

**Comment by pwilkins** [comment]
I'd like eyes on this line: https://github.com/mitodl/PyLmod/pull/11/files#diff-a2ca8308d9281e045010f13ce409eb81R76
get_gradebook_id() is called in base.py.__init__() but properly belongs in the Gradebook class.  Since Gradebook inherits Base, it sounds like a endless loop to have __init__() call it from Base.

**Comment by carsongee** [comment]
Sure, so nothing in base.py uses that property and that initialization should move to the gradebook class `__init__` method which should call it's super `__init__`

File: pylmod/__init__.py [hunk 5]
```diff
@@ -6,20 +5,22 @@
...
         raise DistributionNotFound
 except DistributionNotFound:
     __version__ = 'Please install this project with setup.py'
 else:
-    __version__ = _dist.version
+    __version__ = DIST.version
 
 
-__all__ = ['Client', 'StellarGradeBook', ]
+__all__ = ['Base', 'Client', 'GradeBook', 'Membership']
```
Comment (carsongee) [inline pylmod/__init__.py]: You shouldn't have base here

File: pylmod/base.py [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        self.ses.cert = cert
+        self.ses.timeout = self.TIMEOUT  # connection timeout
+        self.ses.verify = True  # verify site certificate
+
+        log.debug("------------------------------------------------------")
+        log.info("[StellarGradeBook] init base=%s", urlbase)
+        if gbuuid is not None:
+            self.gradebookid = self.get_gradebook_id(gbuuid)
+
+    def get_gradebook_id(self, gbuuid):
```
Comment (carsongee) [inline pylmod/base.py]: This shouldn't be here, this and the extra gbuuid parameter should be in gradebook I think

File: pylmod/membership.py [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+log = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class Membership(Base):  # pylint: disable=too-few-public-methods
+    """
+    Provide API for functions that return group membership data from MIT
+    Learning Modules service.
+    """
+    pass
```
Comment (carsongee) [inline pylmod/membership.py]: I think we probably want to raise a not implemented exception on instantiation

File: pylmod/base.py [hunk 8]
```diff
@@ -0,0 +1,156 @@
+"""
+
```
Comment (carsongee) [inline pylmod/base.py]: what is with all the extra spaces?

File: pylmod/client.py [hunk 9]
```diff
@@ -1,13 +1,14 @@
 """
-Python interface to MIT Learning Module
+Python interface to MIT Learning Modules service
```
Comment (carsongee) [inline pylmod/client.py]: I would say, "Contains the Client class for pylmod that includes all API classes. This should be used if you need to use more than just one LMod API" or similar

File: pylmod/gradebook.py [hunk 10]
```diff
@@ -1,269 +1,33 @@
...
-    sg.create_assignment('midterm1', 'mid1', 1.0, 100.0, '11-04-2013')
-
-    sid, student = sg.get_student_by_email(email)
-    aid, assignment = sg.get_assignment_by_name('midterm1')
-    sg.set_grade(aid, sid, 95.2)
-
-    sg.spreadsheet2gradebook(datafn)
-
+    Provide API for functions that return gradebook-related data from MIT
+    Learning Modules service.
```
Comment (carsongee) [inline pylmod/gradebook.py]: Maybe add a link to the API doc. That would sure be helpful to me

File: pylmod/membership.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+import logging
+from pylmod.base import Base
+
+log = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class Membership(Base):  # pylint: disable=too-few-public-methods
+    """
+    Provide API for functions that return group membership data from MIT
+    Learning Modules service.
```
Comment (carsongee) [inline pylmod/membership.py]: Again, a link to the API doc would be nice.

**Comment by carsongee** [comment]
I'm happy to merge after those last few comments are addressed. What a long strange trip... :snake:

File: pylmod/base.py [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+log = logging.getLogger(__name__)  # pylint: disable=C0103
+
+
+class Base(object):
+    """
+    Python class representing interface to Stellar gradebook.
+
+    Example usage:
+
+    sg = StellarGradeBook('ichuang-cert.pem')
```
Comment (carsongee) [inline pylmod/base.py]: it would be nice to at least correct the class name here.

File: pylmod/base.py [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+    GBUUID = 'STELLAR:/project/mitxdemosite'
+    TIMEOUT = 200  # connection timeout, seconds
+
+    verbose = True
+    gradebookid = None
+
+    def __init__(
+            self,
+            cert,
+            urlbase='https://learning-modules.mit.edu:8443/service/gradebook',
```
Comment (carsongee) [inline pylmod/base.py]: wouldn't this be different in membership and gradebook?  If so, we should probably remove the gradebook here and have the sub classes add in there specific endpoint base.

File: pylmod/__init__.py (hunk 5 above)
Comment (pwilkins) [inline pylmod/__init__.py]: removed.

File: pylmod/base.py (hunk 8 above)
Comment (pwilkins) [inline pylmod/base.py]: removed.

File: pylmod/base.py (hunk 13 above)
Comment (carsongee) [inline pylmod/base.py]: This could be a new issue if you want to get this in

File: pylmod/base.py (hunk 12 above)
Comment (pwilkins) [inline pylmod/base.py]: done.  I also removed "stellargradebook" from all files

File: pylmod/base.py (hunk 13 above)
Comment (pwilkins) [inline pylmod/base.py]: I actually had that code in one my previous commits but took it out during one of the churns.  I'll add the issue and replace the code.

File: pylmod/client.py (hunk 9 above)
Comment (pwilkins) [inline pylmod/client.py]: done.

File: pylmod/membership.py (hunk 7 above)
Comment (carsongee) [inline pylmod/membership.py]: Actually, we really don't, because we couldn't instantiate class

**Pull request merged**